features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/NotificationSettingsPresenter.kt

```diff
@@ -68,6 +76,61 @@ class NotificationSettingsPresenter @Inject constructor(
             observeNotificationSettings(matrixSettings)
         }
 
+        // List of PushProvider -> Distributor
+        val distributors = remember {
+            pushService.getAvailablePushProviders()
+                .flatMap { pushProvider ->
+                    pushProvider.getDistributors().map { distributor ->
+                        pushProvider to distributor
+                    }
+                }
+        }
+        // List of Distributor names
+        val distributorNames = remember {
+            distributors.map { it.second.name }
+        }
+
+        var currentDistributorName by remember { mutableStateOf<AsyncAction<String>>(AsyncAction.Uninitialized) }
+        var refreshPushProvider by remember { mutableIntStateOf(0) }
+
+        LaunchedEffect(refreshPushProvider) {
```

Comment:
Is `refreshPushProvider` really needed? It only changes when the push provider is set, so it seems like it's doing the same job twice.

Reply:
Yes, you're right, probably a remaining of an old version of the code. Will remove.

Reply:
Actually I prefer to remove the line
```
currentDistributorName = AsyncAction.Success(distributor.name)
```
So that we read again the value from the module.

Reply:
It seems a bit more complex, but it should ensure everything is up to date 👍 .

Reply:
3349df5a24084679af5dfd1a668675196011c1c2